librubyfmt/src/parser_state.rs

```diff
@@ -902,11 +902,7 @@ impl BaseParserState {
         if self.heredoc_strings.is_empty() {
             None
         } else {
-            let mut hds = vec![];
-            while let Some(element) = self.heredoc_strings.pop() {
-                hds.push(element);
-            }
-            Some(hds)
+            Some(self.heredoc_strings.drain(..).rev().collect())
```

Comment:
Could also do `std::mem::take` of the `heredoc_strings` and then reverse them, but perhaps it's good to leave the allocation around?

Reply:
I think it's probably a good thing that we leave the memory around for later, but I like the `std::mem::take` suggestion and will keep it in mind for next time!